re.c

```diff
@@ -246,11 +252,16 @@ re_t re_compile(const char* pattern)
   /* 'UNUSED' is a sentinel used to indicate end-of-pattern */
   re_compiled[j].type = UNUSED;
 
+#if defined(RE_ENABLE_MULTI_PATTERNS) && (RE_ENABLE_MULTI_PATTERNS == 1)
   re_p = (re_t)calloc(1, sizeof(re_compiled));
   memcpy(re_p, re_compiled, sizeof(re_compiled));
   return (re_t)re_p;
```

Comment:
``re_p`` is already an instance of ``re_t``, why would anyone need this cast? I'm also unsure about the similar cast below.

Reply:
yes ,delete it.